### PR TITLE
move csi-common to ceph-csi

### PR DESCRIPTION
**Pull request by Madhu-1**
`kubernetes/drivers/csi-common` is no longer maintained. moving common  utility functions to ceph-csi

Signed-off-by: Madhu Rajanna <mrajanna@redhat.com>

### CODE REVIEW

**Comment by Madhu-1** [comment]
@rootfs @gman0 PTAL

File: pkg/csi-common/controllerserver-default.go [hunk 1]
```diff
@@ -18,62 +18,58 @@ package csicommon
...
 func (cs *DefaultControllerServer) CreateSnapshot(ctx context.Context, req *csi.CreateSnapshotRequest) (*csi.CreateSnapshotResponse, error) {
 	return nil, status.Error(codes.Unimplemented, "")
 }
 
+// DeleteSnapshot deletes snapshot
 func (cs *DefaultControllerServer) DeleteSnapshot(ctx context.Context, req *csi.DeleteSnapshotRequest) (*csi.DeleteSnapshotResponse, error) {
 	return nil, status.Error(codes.Unimplemented, "")
 }
 
+// ListSnapshots list snapshosts
```
Comment (gman0) [inline pkg/csi-common/controllerserver-default.go]: listS snapshots

**Comment by gman0** [comment]
No `glog` means this needs an update as well:
https://github.com/ceph/ceph-csi/blob/8e41a8a7eda9651aef5c060257f59b4738e7236d/pkg/util/log.go#L26-L45

**Comment by Madhu-1** [comment]
@gman0 can this be done in followup PR? 

> No `glog` means this needs an update as well:
> [ceph-csi/pkg/util/log.go](https://github.com/ceph/ceph-csi/blob/8e41a8a7eda9651aef5c060257f59b4738e7236d/pkg/util/log.go#L26-L45)
> 
> Lines 26 to 45 in [8e41a8a](/ceph/ceph-csi/commit/8e41a8a7eda9651aef5c060257f59b4738e7236d)
> 
>  // InitLogging initializes klog alongside glog 
>  // XXX: This is just a temporary solution till all deps move to klog 
>  func InitLogging() { 
>  	if err := flag.Set("logtostderr", "true"); err != nil { 
>  		klog.Errorf("failed to set logtostderr flag: %v", err) 
>  		os.Exit(1) 
>  	} 
>   
>  	flag.Parse() 
>   
>  	klogFlags := flag.NewFlagSet("klog", flag.ExitOnError) 
>  	klog.InitFlags(klogFlags) 
>   
>  	// Sync klog flags with glog 
>  	flag.CommandLine.VisitAll(func(f1 *flag.Flag) { 
>  		if f2 := klogFlags.Lookup(f1.Name); f2 != nil { 
>  			f2.Value.Set(f1.Value.String()) // nolint: errcheck, gosec 
>  		} 
>  	}) 
>  }

@gman0 can this be done in followup PR?

**Comment by gman0** [comment]
Could you please also remove https://github.com/ceph/ceph-csi/blob/csi-v1.0/pkg/util/log.go ? Not needed anymore.

**Comment by Madhu-1** [comment]
> Could you please also remove https://github.com/ceph/ceph-csi/blob/csi-v1.0/pkg/util/log.go ? Not needed anymore.

@gman0 is that file still present? i think it  got deleted

**Comment by gman0** [comment]
LGTM

File: cmd/cephfs/main.go [hunk 2]
```diff
@@ -32,8 +33,15 @@ var (
 	metadataStorage = flag.String("metadatastorage", "", "metadata persistence method [node|k8s_configmap]")
 )
 
+func init() {
+	klog.InitFlags(nil)
+	if err := flag.Set("logtostderr", "true"); err != nil {
```
Comment (humblec) [inline cmd/cephfs/main.go]: @Madhu-1 did you validate `logtostderr` works ?

**Comment by humblec** [comment]
@Madhu-1 one quick comment

File: cmd/cephfs/main.go (hunk 2 above)
Comment (Madhu-1) [reply cmd/cephfs/main.go]: @humblec yes tested, am able to see  all  logs

Comment (humblec) [reply cmd/cephfs/main.go]: Thanks!

**Comment by humblec** [comment]
lgtm.

**Comment by Madhu-1** [comment]
resolved merge conflicts

@rootfs PTAL.

File: cmd/cephfs/main.go (hunk 2 above)
Comment (rootfs) [reply cmd/cephfs/main.go]: cc @gman0

File: pkg/cephfs/controllerserver.go [hunk 3]
```diff
@@ -17,16 +17,15 @@ limitations under the License.
 package cephfs
 
 import (
+	csicommon "github.com/ceph/ceph-csi/pkg/csi-common"
```
Comment (rootfs) [inline pkg/cephfs/controllerserver.go]: where is this pkg?

Comment (Madhu-1) [reply pkg/cephfs/controllerserver.go]: ```.../pkg/csi-common/controllerserver-default.go → pkg/csi-common/controllerserver-default.go```

this package is moved under pkg/csi-commom

Comment (rootfs) [reply pkg/cephfs/controllerserver.go]: I see, github is funny

**Pull request merged**